graylog2-server/src/main/java/org/graylog2/lookup/Catalog.java

```diff
@@ -54,37 +63,24 @@ public Entry load(String id) {
                         var catalog = contentPackService.getEntityExcerpts();
                         var excerpt = catalog.get(id);
                         if (excerpt != null) {
-                            return new Entry(excerpt.type().name(), excerpt.title());
+                            if (excerpt.title() != null) {
+                                return new Entry(excerpt.title());
+                            } else {
+                                return new Entry(id);
+                            }
                         } else {
-                            return new Entry("Unknown entity: " + id, "Unknown entity: " + id);
+                            return Entry.UNKNOWN_ENTITY;
                         }
                     }
                 });
     }
 
-    public String getTitle(final GRN grn) {
+    public Entry getEntry(final GRN grn) {
```

Comment:
Maybe we could let this return `Optional<Entry>` and turn the `title` attribute of the `Entry` record into an `Optional<String>` as well? This way we could properly reflect the state of the lookup:

  - `getEntry` returns empty optional: No entry in the catalog
  - `getEntry` returns `Entry` record with `title` being empty optional: Entry is present, but entity has no title
  - `getEntry` returns `Entry` record with `title` being set: Entry is present and has title

Reply:
That would complicate ALL places where the catalog is used so far. Every invocation would have to check for those 3 possible states and make decision about what to do with it. 
If you agree, I'd leave it as it is because of that.

Reply:
I am seeing three usages of `Catalog#getEntry` at the moment, so I think it is a good time to make the change before we adopt it in more places.

Reply:
Ok, I will generalize Catalog if you believe it may be used in more places.
I will add a small util class that will encapsulate the logic that is now used in those 3 places where catalog is used.